test/test_client.py

```diff
@@ -0,0 +1,77 @@
+from collections import Counter
+from unittest import TestCase
+import logging
+
+from mock import Mock, patch
+
+from apns2.client import APNsClient, CONCURRENT_STREAMS_SAFETY_MAXIMUM
+from apns2.payload import Payload
+
+class ClientTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        logging.basicConfig()
+        cls.tokens = ["%064x" % i for i in range(10000)]
+        cls.notification = Payload(alert="Hello, world!")
+        cls.topic = "com.example.App"
+
+    
+    def setUp(self):
+        self.open_streams = 0
+        self.max_open_streams = 0
+        self.mock_results = None
+        
+        with patch("apns2.client.HTTP20Connection") as mock_connection_constructor, \
+            patch("apns2.client.init_context"):
+            self.mock_connection = Mock()
+            self.mock_connection.get_response.side_effect = self.mock_get_response
```

Comment:
Why would I want to do that? My reasons for using the Mock library are:

1. Inline, one-line generation of mocks. Note that I have more than one mocked class here. I would need at least three classes: MockConnection, MockSettings, MockResponse.
2. Mock gives me useful things like call_count, assert_called_with, side_effect.
3. mock.patch is very useful, with automatic undo of the patch when finished.